power_grid_model_c/power_grid_model/include/power_grid_model/main_core/input.hpp

```diff
@@ -124,6 +127,31 @@ inline void add_component(MainModelState<ComponentContainer>& state, ForwardIter
                 }
             }();
 
+            if (regulated_object_idx.group == get_component_type_index<Transformer>(state)) {
```

Comment:
yeah this entire function needs refactoring. i guess it's ok for now (since it's already a wretch)